tests/src/testpathfollowing.cpp

```diff
@@ -39,133 +38,139 @@ static auto stiffnessMatrix(const Eigen::VectorXd& D, [[maybe_unused]] double la
   return mat;
 }
 
-static auto simple2DOperatorArcLengthTest() {
-  double lambda = 0;
-  Eigen::VectorXd D;
-  D.setZero(2);
-
-  auto fvLambda  = [&](auto&& D_, auto&& lambda_) { return residual(D_, lambda_); };
-  auto dfvLambda = [&](auto&& D_, auto&& lambda_) { return stiffnessMatrix(D_, lambda_); };
-
-  auto nonLinOp = Ikarus::NonLinearOperator(Ikarus::functions(fvLambda, dfvLambda), Ikarus::parameter(D, lambda));
-
+template <typename NonLinearOperator>
+static auto simple2DOperatorArcLengthTest(NonLinearOperator& nonLinOp, double stepSize, int loadSteps) {
+  resetNonLinearOperatorParametersToZero(nonLinOp);
   auto linSolver = Ikarus::LinearSolver(Ikarus::SolverTypeTag::d_LDLT);
+  auto pft       = Ikarus::StandardArcLength{};  // Type of path following technique
 
-  double stepSize = 0.1;
-  int load_steps  = 50;
-
-  auto pft = Ikarus::StandardArcLength{};  // Path following type
-
-  static_assert(PathFollowingStrategy<decltype(pft), decltype(nonLinOp)>,
-                "StandardArcLength is a PathFollowingStrategy");
+  auto nr  = Ikarus::makeNewtonRaphsonWithSubsidiaryFunction(nonLinOp, std::move(linSolver));
+  auto alc = Ikarus::PathFollowing(nr, loadSteps, stepSize, pft);
 
-  auto nr                      = Ikarus::makeNewtonRaphsonWithSubsidiaryFunction(nonLinOp, std::move(linSolver));
-  auto alc                     = Ikarus::PathFollowing(nr, load_steps, stepSize, pft);
   auto nonLinearSolverObserver = std::make_shared<Ikarus::NonLinearSolverLogger>();
+  auto pathFollowingObserver   = std::make_shared<Ikarus::ControlLogger>();
   nr->subscribeAll(nonLinearSolverObserver);
-  const auto controlInfo = alc.run();
+  alc.subscribeAll(pathFollowingObserver);
+  const auto controlInfo              = alc.run();
+  std::vector<int> expectedIterations = {1, 3, 3, 3, 3};
+  Eigen::VectorXd expectedDisplacement;
+  expectedDisplacement.setZero(2);
+  expectedDisplacement << 0.0883524725970593, 0.3486891582376427;
+  double expectedLambda = 0.4877655288280236;
 
   TestSuite t("Arc Length with Subsidiary function");
-  t.check(controlInfo.success, "Successful result");
+  t.check(controlInfo.success, "No convergence");
+  for (auto i = 0; i < 2; ++i)
+    checkScalars(t, nonLinOp.firstParameter()[i], expectedDisplacement[i], " --> " + pft.name);
```

Comment:
can#T you do here `isApprixSame`and also use the testSuite? 

As 

```cpp
t.check(isApproxSame(nonLinOp.firstParameter(),expectedDisplacement));
```

Reply:
`checkScalars` was created also to hide the `cout` statements needed if the test fails. More refactoring of this interface will be done in a separate PR which will linked to the issue #212 . It is not done in this PR as touching these functions will also touch several other files while refactoring and it can then go beyond the scope of this PR.